crates/fluvio-protocol-derive/src/util.rs

```diff
@@ -1,75 +1,90 @@
+                    ),
+                ))
+            }
+        }
+        _ => Err(syn::Error::new_spanned(
+            value,
+            format!("Expected {attr_name} attribute to be a Lit: `{attr_name} = \"...\"`"),
+        )),
+    }
+}
 
 pub(crate) fn find_attr(attrs: &[Attribute], name: &str) -> Option<Meta> {
     attrs.iter().find_map(|a| {
-        if let Ok(meta) = a.parse_meta() {
-            if meta.path().is_ident(name) {
-                Some(meta)
-            } else {
-                //println!("attr name: {}",meta.name());
-                None
-            }
+        if a.meta.path().is_ident(name) {
+            Some(a.meta.clone())
         } else {
-            //println!("unrecog attribute");
             None
         }
     })
 }
-
-pub(crate) fn find_name_attribute<'a>(meta: &'a Meta, name: &str) -> Option<&'a MetaNameValue> {
-    find_meta(meta, name).map(|meta| match meta {
-        Meta::NameValue(name_value) => name_value,
-        _ => panic!("should not happen"),
-    })
-}
-
-pub(crate) fn find_meta<'a>(meta: &'a Meta, name: &str) -> Option<&'a Meta> {
-    if let Meta::List(list) = meta {
-        for attr in list.nested.iter() {
-            if let NestedMeta::Meta(named_meta) = attr {
-                match named_meta {
-                    Meta::NameValue(meta_name_value) => {
-                        if meta_name_value.path.is_ident(name) {
-                            return Some(named_meta);
-                        }
-                    }
-                    Meta::Path(path) => {
-                        if path.is_ident(name) {
-                            return Some(named_meta);
-                        }
-                    }
-                    Meta::List(_) => {}
-                }
+pub(crate) fn find_name_value_from_meta_list(
+    meta_list: &MetaList,
+    attr_name: &str,
+) -> Option<MetaNameValue> {
+    for item in meta_list
+        .parse_args_with(Punctuated::<syn::Meta, Token![,]>::parse_terminated)
+        .unwrap_or_else(|err| panic!("Invalid #[new] attribute: {}", err))
+    {
+        if let Meta::NameValue(kv) = &item {
+            if kv.path.is_ident(attr_name) {
+                return Some(kv.clone());
             }
-        }
+        };
     }
-
     None
 }
 
-/// find name value with integer value
-pub(crate) fn find_int_name_value(version_meta: &Meta, attr_name: &str) -> Option<u64> {
-    if let Some(attr) = find_name_attribute(version_meta, attr_name) {
-        match &attr.lit {
-            Lit::Int(version_val) => {
-                //  println!("version value: {}",version_val.value());
-                version_val.base10_parse::<u64>().ok()
-            }
-            _ => unimplemented!(),
-        }
-    } else {
-        None
+pub(crate) fn find_name_value_from_meta(meta: &Meta, attr_name: &str) -> Option<MetaNameValue> {
+    let mut result_meta: Option<MetaNameValue> = None;
```

Comment:
Not critical, but if you have time, remove this mut variable as well.

Reply:
Opps i missed that.